Model/SourcesRepository.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Ampersand\DisableStockReservation\Model;
+
+use Braintree\Exception;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources;
+use Ampersand\DisableStockReservation\Api\Data\SourcesInterface;
+use Ampersand\DisableStockReservation\Api\SourcesRepositoryInterface;
+use Ampersand\DisableStockReservation\Api\Data\SourcesInterfaceFactory;
+use Ampersand\DisableStockReservation\Model\Sources as SourcesModel;
+
+/**
+ * Class SourcesRepository
+ * @package Ampersand\DisableStockReservation\Model
+ */
+class SourcesRepository implements SourcesRepositoryInterface
+{
+    /**
+     * @var SourcesInterfaceFactory
+     */
+    protected $sourcesFactory;
+
+    /**
+     * @var Sources
+     */
+    protected $sourcesResourceModel;
+
+    /**
+     * SourcesRepository constructor.
+     * @param SourcesInterfaceFactory $sourcesFactory
+     * @param Sources $sourcesResourceModel
+     */
+    public function __construct(
+        SourcesInterfaceFactory $sourcesFactory,
+        Sources $sourcesResourceModel
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourcesInterface
+     * @throws \Exception
+     */
+    public function getByOrderId(string $orderId): SourcesInterface
+    {
+        /** @var SourcesInterface $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
```

Comment:
Does PHPStan still fail if we change the docblock here to the sources model instead of the interface? Using the `instanceof` workaround might be excessive on second thoughts